### PR TITLE
Use dashboard to restart the workspace

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

### What does this PR do?
Adds an event handler to the Dashboard to restart the workspace by the request from the IDE opened in the frame.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/15006

Connected with https://github.com/eclipse/che-theia/pull/541

### CODE REVIEW

**Review by benoitf — APPROVED** [review]

File: dashboard/src/components/api/workspace/che-workspace.factory.ts [hunk 1]
```diff
@@ -407,6 +407,34 @@ export class CheWorkspace {
...
+        token: token
+      }
+    });
+
+    promise.then((response: ng.IHttpPromiseCallbackArg<che.IWorkspace>) => {
+      defer.resolve();
+    }, (error: any) => {
+      if (error && error.status === 304) {
+        defer.resolve();
+        return;
```
Comment (AndrienkoAleksandr) [inline dashboard/src/components/api/workspace/che-workspace.factory.ts:430]: Should we return `return defer.promise;`?

**Review by AndrienkoAleksandr — APPROVED** [review]

File: dashboard/src/components/api/workspace/che-workspace.factory.ts (hunk 1 above)
Comment (akurinnoy) [reply dashboard/src/components/api/workspace/che-workspace.factory.ts:430]: No, we only need there to resolve or reject that deferred object promise. And return it at the end of the method.

**Review by akurinnoy — APPROVED** [review]

**Review by evidolob — APPROVED** [review]

**Comment by evidolob** [comment]
@vitaliy-guliy you need to cherry-pick this PR to `7.3.x` branch

**Comment by vitaliy-guliy** [comment]
Corresponding PR to 7.3.x https://github.com/eclipse/che/pull/15194

**Comment by vitaliy-guliy** [comment]
ci-test

**Pull request merged**